library/src/com/google/maps/android/clustering/ClusterManager.java

```diff
@@ -92,13 +93,23 @@ public void setAlgorithm(Algorithm<T> algorithm) {
             if (mAlgorithm != null) {
                 algorithm.addItems(mAlgorithm.getItems());
             }
-            mAlgorithm = new PreCachingAlgorithmDecorator<T>(algorithm);
+
+            mAlgorithm = algorithm;
         } finally {
             mAlgorithmLock.writeLock().unlock();
         }
+
+        if (mAlgorithm instanceof GoogleMap.OnCameraChangeListener) {
+            ((GoogleMap.OnCameraChangeListener) mAlgorithm).onCameraChange(mMap.getCameraPosition());
+        }
+
         cluster();
     }
 
+    public void setClusterOnlyVisibleArea(boolean onlyVisibleArea) {
```

Comment:
Ahh, good point. My main concern is that the behaviour of using `setClusterOnlyVisibleArea` on anything other than the new Algorithm you've introduced is undefined.

The only other option available is to break the Algorithm interface by adding the method suggested above. I've thought about this a bit and I think that it's the right thing to do but I (or whoever releases the next version to Maven) will need to make sure we bump the version number appropriately so it's clear what's happening.

What do you think about that? Reasonable? Or too drastic?